theme/theme.inc

```diff
@@ -229,6 +229,53 @@ function islandora_solr_metadata_query_fields($object, &$solr_fields) {
         unset($field);
       }
 
+      // After values are omitted/dedupped filter down the displayed results.
+      foreach ($solr_fields as &$field) {
+        // Check if the field is configured to filter based on whole field.
+        if (($truncation = drupal_array_get_nested_value($field, array('truncation'))) && $truncation['max_length'] > 0 && $truncation['truncation_type'] == 'whole_field_option') {
+          if (count($field['value']) > 0) {
+            $mod_path = drupal_get_path('module', 'islandora_solr_metadata');
+            drupal_add_js("$mod_path/js/truncation-toggle.js");
+
+            // Build two arrays for display, one filtered by size (based on
+            // character count of what is displayed in the field) and one with
+            // all the data.
+            $character_output_count = 0;
+            $truncated = FALSE;
+            foreach ($field['value'] as $index => $current_value) {
+              $list_full[] = $current_value;
+              if ($character_output_count + strlen($current_value) <= $truncation['max_length']) {
```

Comment:
Same business with the `strlen()` calls here as in https://github.com/Islandora/islandora_solr_search/pull/233